### PR TITLE
Support of readOnly and writeOnly with anyOf/oneOf/allOf

### LINKED ISSUE #31: Support of readOnly and writeOnly with anyOf/oneOf/allOf
Currently when defining objects with oneOf, anyOf or allOf both readOnly and writeOnly logic doesn't work at all.

Example:
```yaml
    AnyOf:
      anyOf:
        - $ref: '#/components/schemas/User'
        - type: object
          required:
            - additionalOneOfField
          properties:
            additionalOneOfField:
              type: string
```

### PR DESCRIPTION
**Pull request by kobik**
Closes #31

### CODE REVIEW

File: src/utils/schemaUtils.js [hunk 1]
```diff
@@ -33,7 +33,14 @@ function getAllResponseContentTypes(responses) {
  * @param {string} omitByValue omit if the prop value equals to this value
  */
 function omitPropsFromSchema(dereferencedSchema, omitByPropName, omitByValue) {
-    if (dereferencedSchema.type === 'object') {
+    const combinedSchemaType = getCombinedSchemaType(dereferencedSchema);
```
Comment (kobik) [inline src/utils/schemaUtils.js]: I'd be glad for suggestions for a better name for `combinedSchemaType` and `getCombinedSchemaType`

File: test/openapi3/general/readOnlyWriteOnly.yaml [hunk 2]
```diff
@@ -33,19 +33,19 @@ paths:
...
               schema:
                 $ref: '#/components/schemas/UserOptional'
-  /users/utf8:
+  /users/hal:
     post:
       requestBody:
         required: true
         content:
-          application/json;utf8:
+          application/hal+json:
```
Comment (idanto) [inline test/openapi3/general/readOnlyWriteOnly.yaml]: Why did you change it to hal?

Comment (kibertoad) [reply test/openapi3/general/readOnlyWriteOnly.yaml]: I think that utf8 part should never be part of API spec, instead we should handle requests that include it in header as just application/json
Support for that is implemented here: https://github.com/Zooz/api-schema-builder/pull/29
